src/codegen/codegen_cuda.cc

```diff
@@ -50,6 +50,8 @@ void CodeGenCUDA::AddFunction(LoweredFunc f) {
 std::string CodeGenCUDA::Finish() {
   if (enable_fp16_) {
     decl_stream << "#include <cuda_fp16.h>\n";
+    decl_stream << "__device__ half max(const half a, const half b)\n"
```

Comment:
Do we know which operators we have to overload as such? "max" is one of them. Do we need others?

Reply:
For now, I only find `max` that need to be overloaded.

BTW, I have a question about the checks.
Why this commit cannot be built today? It was successful yesterday.

Reply:
Hi, we saw more failures while trying to run full resnet

https://github.com/dmlc/tvm/issues/3816#issuecomment-524067090

I think, we are missing all reduce ops. Will it be possible for you to help with this? (In a separate PR, this one is good to go)